samples/algorithms/repeat-until-success/RepeatUntilSuccess.qs

```diff
@@ -0,0 +1,116 @@
+    ///
+    /// The program executes a circuit on a "target" qubit using an "auxiliary" and 
+    /// "resource" qubit. The circuit consists of two parts (red and blue in image).
+    /// The goal is to measure Zero for both the auxiliary and resource qubit.
+    /// If this succeeds, the program will have effectively applied an 
+    /// Rz(arctan(2)) gate (also known as V_3 gate) on the target qubit.
+    /// If this fails, the program reruns the circuit up to <limit> times.
+    @EntryPoint()
+    operation ApplyRzArcTan2(
+        inputValue : Bool,
+        inputBasis: Pauli,
+        limit: Int
+    ) : (Bool, Result, Int) {
+        using ((auxiliary, resource, target) = (Qubit(), Qubit(), Qubit())) {
+            /// Prepare auxiliary and resource qubits in |+> state
+            SetXZeroFromOne(auxiliary);
+            SetXZeroFromOne(resource);
+            /// Prepare target qubit in |0> or |1> state, depending on input value
+            PrepareValueForBasis(inputValue, inputBasis, target);
+
+            /// Initialize results to One by default.
+            mutable done = false;
+            mutable success = false;
+            mutable numIter = 0;
+
+            repeat {
+                SetXZeroFromOne(auxiliary);
+                // Run Part 1 of the program.
+                let result1 = ApplyAndMeasurePart1(auxiliary, resource);
+                // We'll only run Part 2 if Part 1 returns Zero.
+                // Otherwise, we'll skip and rerun Part 1 again.
+                if (result1 == Zero) { //|0X>
+                    let result2 = ApplyAndMeasurePart2(resource, target);
+                    if (result2 == Zero) { //|00>
+                        set success = true;
+                    } else { //|01>
+                        H(auxiliary); // Reset auxiliary from |0> to |+>
+                        SetXZeroFromOne(resource); // Reset resource from |1> to |+>
+                        Adjoint Z(target); // Correct effective Z rotation on target
+                    }
+                } else { // |1X>, skip Part 2
+                    // Reset auxiliary from |1> to |+>
+                    SetXZeroFromOne(auxiliary);
+                }
+                set done = (success or numIter >= limit);
+                set numIter = numIter + 1;
+            }
+            until (done);
+
+            let result = Measure([inputBasis], [target]);
+            return ( success, result, numIter );
+        }
+    }
+
+    /// Prepare qubit in either |0> or |1> for the given basis
+    operation PrepareValueForBasis(
+        inputValue : Bool,
+        inputBasis : Pauli,
+        input: Qubit
+        ) : Unit {
+            if (inputValue) {
+                X(input);
+            }
+            PrepareQubit(inputBasis, input);
+    }
+
+    /// Prepare qubit in |+> state given it is in the |1> state
+    operation SetXZeroFromOne(target : Qubit) : Unit {
+        X(target); // Flip to |0>
```

Comment:
```suggestion
        AssertQubit(One, target);
        X(target); // Flip to |0〉
```

I suggest to add a precondition that checks whether `target` is in expected state |1〉